src/menus/StsMenu/StsValidityLayerSwitcher.js

```diff
@@ -4,22 +4,39 @@ import { useDispatch, useSelector } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { FormGroup, FormControlLabel } from '@material-ui/core';
 import { unByKey } from 'ol/Observable';
-import SBBSwitch from '../../components/SBBSwitch/SBBSwitch';
-import { highlights, gttos, premium } from '../../config/ch.sbb.sts';
+import SBBSwitch from '../../components/SBBSwitch';
+// import { highlights, gttos, premium } from '../../config/ch.sbb.sts';
+import { STS_KEY } from '../../utils/constants';
 import { setFeatureInfo } from '../../model/app/actions';
 
+const stsStyleLayerRegex = new RegExp(
+  `^${STS_KEY}.validity.(highlights|premium|gttos)$`,
+);
+const stsHightlightLayerRegex = new RegExp(`^${STS_KEY}.validity.highlights$`);
 function StsValidityLayerSwitcher() {
   const { t } = useTranslation();
   const dispatch = useDispatch();
   const [revision, forceRender] = useState();
   const layers = useSelector((state) => state.map.layers);
+  const stsLayers = useMemo(() => {
+    return layers?.filter((layer) => stsStyleLayerRegex.test(layer.key));
+  }, [layers]);
+  const [layersVisible, setLayersVisible] = useState(
+    stsLayers.filter((l) => l.visible).map((l) => l.key),
+  );
```

Comment:
since layersVisible is probably change on each render (l33)  it's safe to define it outside a useXXX.

const layersVisible = stsLayers.filter((l) => l.visible).map((l) => l.key);

It avoids some code in the change's visible callback. if you prefer you can use a 
useMemo(()=> {...}, [stsLayer,revision])

Reply:
Yeah, I guess putting it in state is unnecessary